pysat/instruments/icon_mighti.py

```diff
@@ -259,43 +259,58 @@ def clean(inst, clean_level=None):
 
     """
 
-    if inst.tag[0:2] == 've':
+    if inst.tag.find('los') >= 0:
+        # dealing with LOS winds
+        var = 'Line_of_Sight_Wind'
+        if 'Wind_Quality' in inst.variables:
+            qual_flag = 'Quality_Flags'
+        else:
+            qual_flag = 'ICON_L21_Quality_Flags'
+            var = 'ICON_L21_' + var
+        if inst.clean_level in ['clean', 'dusty']:
+            # find location with any of the flags set
+            idx, idy, = np.where(inst[qual_flag].any(axis=2))
+            inst[idx, idy, var] = np.nan
+        else:
+            # dirty and worse lets everything through
+            pass
+
+    elif inst.tag.find('vector') >= 0:
         # vector winds area
-        mvars = ['Zonal_Wind', 'Meridional_Wind']
-        if clean_level == 'good':
-            idx, = np.where(inst['Wind_Quality'] != 1)
-            inst[idx, mvars] = np.nan
-        elif clean_level == 'dusty':
-            idx, = np.where(inst['Wind_Quality'] < 0.5)
-            inst[idx, mvars] = np.nan
+        vars = ['Zonal_Wind', 'Meridional_Wind']
+        if 'Wind_Quality' in inst.variables:
+            qual_flag = 'Wind_Quality'
+        else:
+            qual_flag = 'ICON_L22_Wind_Quality'
+            vars = ['ICON_L22_' + x for x in vars]
+        if inst.clean_level == 'clean':
+            for var in vars:
+                inst[var] = inst[var].where(inst[qual_flag] == 1.0)
```

Comment:
Found a fix: using xarray's built in where function to mask `False` values as nans.  There is probably a much deeper bug in how we deal with setting 2d data, but I would rather deal with that in one of the xarray issues.